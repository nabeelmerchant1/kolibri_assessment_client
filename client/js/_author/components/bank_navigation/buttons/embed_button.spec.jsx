import React             from 'react';
import { shallow }       from 'enzyme';
import EmbedButton       from './embed_button';

describe('embed_button component', () => {
  let props;
  let result;
  let focused;

  beforeEach(() => {
    focused = false;
    props = {
      assessment: {
        isPublished: true,
        assessOffered: false
      },
      getEmbedCode: () => {},
      onFocus: () => { focused = true; },
      baseEmbedUrl: '',
    };
    result = shallow(<EmbedButton {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the onFocus function', () => {
    expect(focused).toBeFalsy();
    result.find('button').simulate('focus');
    expect(focused).toBeTruthy();
  });

  it('renders an input when assessOffered', () => {
    props.assessment.assessOffered = true;
    result = shallow(<EmbedButton {...props} />);
    expect(result.find('input')).toBeDefined();
  });

  it('includes the unlock_prev always setting in the embed code', () => {
    props.assessment.assessmentOffered = [{
      unlockPrevious: 'ALWAYS'
    }];
    result = shallow(<EmbedButton {...props} />);
    expect(result.find('#embedInput').html()).toContain('&amp;unlock_prev=ALWAYS');
  });

  it('includes the unlock_prev never setting in the embed code', () => {
    props.assessment.assessmentOffered = [{
      unlockPrevious: 'NEVER'
    }];
    result = shallow(<EmbedButton {...props} />);
    expect(result.find('#embedInput').html()).toContain('&amp;unlock_prev=NEVER');
  });

  it('does not modify the embed code if no unlockPrevious value', () => {
    props.assessment.assessmentOffered = [{
      foo: 'bar'
    }];
    result = shallow(<EmbedButton {...props} />);
    expect(result.find('#embedInput').html()).not.toContain('&amp;unlock_prev');
  });

  it('includes a title attribute in the embed code', () => {
    props.assessment.assessmentOffered = [{
      foo: 'bar'
    }];
    props.assessment.displayName = {
      text: 'test'
    };
    result = shallow(<EmbedButton {...props} />);
    expect(result.find('#embedInput').html()).toContain('title=&quot;test Assessment&quot;');
  });
});
