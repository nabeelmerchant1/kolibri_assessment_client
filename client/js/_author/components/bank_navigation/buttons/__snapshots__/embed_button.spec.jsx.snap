// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`embed_button component matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="au-c-btn au-c-btn--sm au-c-btn--table"
    onClick={[Function]}
    onFocus={[Function]}
>
    embed code
</button>,
  "nodes": Array [
    <button
      className="au-c-btn au-c-btn--sm au-c-btn--table"
      onClick={[Function]}
      onFocus={[Function]}
>
      embed code
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <EmbedButton
        assessment={
                Object {
                        "assessOffered": false,
                        "isPublished": true,
                      }
        }
        baseEmbedUrl=""
        getEmbedCode={[Function]}
        onFocus={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "assessment": Object {
            "assessOffered": false,
            "isPublished": true,
          },
          "baseEmbedUrl": "",
          "getEmbedCode": [Function],
          "onFocus": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="au-c-btn au-c-btn--sm au-c-btn--table"
          onClick={[Function]}
          onFocus={[Function]}
>
          embed code
</button>,
        "_debugID": 2,
        "_renderedOutput": <button
          className="au-c-btn au-c-btn--sm au-c-btn--table"
          onClick={[Function]}
          onFocus={[Function]}
>
          embed code
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <EmbedButton
    assessment={
        Object {
            "assessOffered": false,
            "isPublished": true,
          }
    }
    baseEmbedUrl=""
    getEmbedCode={[Function]}
    onFocus={[Function]}
/>,
}
`;
